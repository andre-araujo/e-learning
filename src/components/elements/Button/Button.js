import React from 'react';
import { node, number, bool } from 'prop-types';

import {
  StyledButton,
  Text,
} from './styles';

function Button({
  children,
  secondary,
  ...props
}) {
  return (
    <StyledButton secondary={secondary} {...props}>
      <Text>
        {children}
      </Text>
    </StyledButton>
  );
}

Button.defaultProps = {
  children: '',
  secondary: false,
  full: false,
  gap: 0,
};

Button.propTypes = {
  children: node,
  secondary: bool,
  full: bool,
  gap: number,
};

export default Button;
