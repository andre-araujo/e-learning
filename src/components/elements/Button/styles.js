import styled, { css } from 'styled-components';

export const StyledButton = styled.button`
  border: 2px solid #27c2d5;
  height: 40px;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  padding : 0 20px;
  outline: none;
  cursor: pointer;
  transition:
    .2s color,
    .2s background,
    .2s transform;

  ${({ full }) => full && css`
    width: 100%;
  `}

  ${({ secondary }) => secondary && css`
    color: #27c2d5;
    background: white;
    :active,
    :hover {
      opacity: 0.9
    }
  `}

  ${({ secondary }) => !secondary && css`
    color: white;
    background: #27c2d5;
    :active,
    :hover {
      opacity: 0.9
    }
  `}

  :active {
    transform: translateY(-3px);
  }

  ${({ gap }) => gap && css`
    margin-right: ${gap}px;
  `};
`;

export const Text = styled.span`
  font-weight: 800;
  font-size: 12px;
  font-family: Roboto;

  :first-letter {
    text-transform: uppercase;
  }
`;
