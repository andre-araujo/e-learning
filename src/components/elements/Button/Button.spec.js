import React from 'react';
import { mount } from 'enzyme';

import Button from './Button';

describe('Button component', () => {
  it('should match snapshot', () => {
    expect(mount(<Button />)).toMatchSnapshot();
  });
});
