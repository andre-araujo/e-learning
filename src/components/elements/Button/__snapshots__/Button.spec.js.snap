// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should match snapshot 1`] = `
<styled.button
  disabled={false}
  full={false}
  gap={0}
  minWidth={null}
  secondary={false}
>
  <styled.span />
</styled.button>
`;
